# Disease Outbreak Early Warning System

## Notebook 01: Data Collection and Integration

### Purpose

The objective of this notebook is to collect, inspect, and integrate dengue surveillance data for subsequent time series analysis and outbreak forecasting.

### Tasks

* Load dengue feature data
* Load dengue case count data
* Merge surveillance and environmental datasets
* Inspect data quality
* Assess missing values
* Prepare the dataset for exploratory analysis

### Expected Outcome

A unified dengue surveillance dataset containing epidemiological and environmental variables that can be used for outbreak detection and forecasting.



Import Libraries


In [1]:
import pandas as pd
import numpy as np

pd.set_option("display.max_columns", None)

print("Libraries imported successfully.")

Libraries imported successfully.


## Loading the Dataset

The dengue forecasting dataset consists of environmental measurements and weekly dengue case counts.

The environmental variables include rainfall, humidity, temperature, and vegetation indices that may influence disease transmission dynamics.


In [2]:
features = pd.read_csv(
    "/content/dengue_features_train.csv"
)

labels = pd.read_csv(
    "/content/dengue_labels_train.csv"
)

print("Features Shape:", features.shape)
print("Labels Shape:", labels.shape)

Features Shape: (1456, 24)
Labels Shape: (1456, 4)


## Inspecting the Data Structure

Before merging the datasets, it is important to understand the available variables and verify that the datasets contain the expected observations.


In [3]:
features.head()

,city,year,weekofyear,week_start_date,ndvi_ne,ndvi_nw,ndvi_se,ndvi_sw,precipitation_amt_mm,reanalysis_air_temp_k,reanalysis_avg_temp_k,reanalysis_dew_point_temp_k,reanalysis_max_air_temp_k,reanalysis_min_air_temp_k,reanalysis_precip_amt_kg_per_m2,reanalysis_relative_humidity_percent,reanalysis_sat_precip_amt_mm,reanalysis_specific_humidity_g_per_kg,reanalysis_tdtr_k,station_avg_temp_c,station_diur_temp_rng_c,station_max_temp_c,station_min_temp_c,station_precip_mm
0,sj,1990,18,1990-04-30,0.122600,0.103725,0.198483,0.177617,12.42,297.572857,297.742857,292.414286,299.8,295.9,32.00,73.365714,12.42,14.012857,2.628571,25.442857,6.900000,29.4,20.0,16.0
1,sj,1990,19,1990-05-07,0.169900,0.142175,0.162357,0.155486,22.82,298.211429,298.442857,293.951429,300.9,296.4,17.94,77.368571,22.82,15.372857,2.371429,26.714286,6.371429,31.7,22.2,8.6
2,sj,1990,20,1990-05-14,0.032250,0.172967,0.157200,0.170843,34.54,298.781429,298.878571,295.434286,300.5,297.3,26.10,82.052857,34.54,16.848571,2.300000,26.714286,6.485714,32.2,22.8,41.4
3,sj,1990,21,1990-05-21,0.128633,0.245067,0.227557,0.235886,15.36,298.987143,299.228571,295.310000,301.4,297.0,13.90,80.337143,15.36,16.672857,2.428571,27.471429,6.771429,33.3,23.3,4.0
4,sj,1990,22,1990-05-28,0.196200,0.262200,0.251200,0.247340,7.52,299.518571,299.664286,295.821429,301.9,297.5,12.20,80.460000,7.52,17.210000,3.014286,28.942857,9.371429,35.0,23.9,5.8


In [4]:
labels.head()

,city,year,weekofyear,total_cases
0,sj,1990,18,4
1,sj,1990,19,5
2,sj,1990,20,4
3,sj,1990,21,3
4,sj,1990,22,6


## Dataset Integration

The environmental features and dengue case counts are merged using the common identifiers:

* City
* Year
* Epidemiological Week

This creates a single dataset for analysis.


In [5]:
dengue_df = pd.merge(
    features,
    labels,
    on=[
        "city",
        "year",
        "weekofyear"
    ]
)

dengue_df.head()

,city,year,weekofyear,week_start_date,ndvi_ne,ndvi_nw,ndvi_se,ndvi_sw,precipitation_amt_mm,reanalysis_air_temp_k,reanalysis_avg_temp_k,reanalysis_dew_point_temp_k,reanalysis_max_air_temp_k,reanalysis_min_air_temp_k,reanalysis_precip_amt_kg_per_m2,reanalysis_relative_humidity_percent,reanalysis_sat_precip_amt_mm,reanalysis_specific_humidity_g_per_kg,reanalysis_tdtr_k,station_avg_temp_c,station_diur_temp_rng_c,station_max_temp_c,station_min_temp_c,station_precip_mm,total_cases
0,sj,1990,18,1990-04-30,0.122600,0.103725,0.198483,0.177617,12.42,297.572857,297.742857,292.414286,299.8,295.9,32.00,73.365714,12.42,14.012857,2.628571,25.442857,6.900000,29.4,20.0,16.0,4
1,sj,1990,19,1990-05-07,0.169900,0.142175,0.162357,0.155486,22.82,298.211429,298.442857,293.951429,300.9,296.4,17.94,77.368571,22.82,15.372857,2.371429,26.714286,6.371429,31.7,22.2,8.6,5
2,sj,1990,20,1990-05-14,0.032250,0.172967,0.157200,0.170843,34.54,298.781429,298.878571,295.434286,300.5,297.3,26.10,82.052857,34.54,16.848571,2.300000,26.714286,6.485714,32.2,22.8,41.4,4
3,sj,1990,21,1990-05-21,0.128633,0.245067,0.227557,0.235886,15.36,298.987143,299.228571,295.310000,301.4,297.0,13.90,80.337143,15.36,16.672857,2.428571,27.471429,6.771429,33.3,23.3,4.0,3
4,sj,1990,22,1990-05-28,0.196200,0.262200,0.251200,0.247340,7.52,299.518571,299.664286,295.821429,301.9,297.5,12.20,80.460000,7.52,17.210000,3.014286,28.942857,9.371429,35.0,23.9,5.8,6


In [6]:
print(
    "Merged Dataset Shape:",
    dengue_df.shape
)

Merged Dataset Shape: (1456, 25)


## Data Quality Assessment

Missing values are common in epidemiological and environmental surveillance systems.

Understanding the extent of missingness helps guide preprocessing decisions in later stages of the project.


In [7]:
missing_values = (
    dengue_df.isnull()
    .sum()
    .sort_values(
        ascending=False
    )
)

missing_values[
    missing_values > 0
]

,0
ndvi_ne,194
ndvi_nw,52
station_avg_temp_c,43
station_diur_temp_rng_c,43
ndvi_se,22
ndvi_sw,22
station_precip_mm,22
station_max_temp_c,20
station_min_temp_c,14
precipitation_amt_mm,13


In [8]:
dengue_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1456 entries, 0 to 1455
Data columns (total 25 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   city                                   1456 non-null   object 
 1   year                                   1456 non-null   int64  
 2   weekofyear                             1456 non-null   int64  
 3   week_start_date                        1456 non-null   object 
 4   ndvi_ne                                1262 non-null   float64
 5   ndvi_nw                                1404 non-null   float64
 6   ndvi_se                                1434 non-null   float64
 7   ndvi_sw                                1434 non-null   float64
 8   precipitation_amt_mm                   1443 non-null   float64
 9   reanalysis_air_temp_k                  1446 non-null   float64
 10  reanalysis_avg_temp_k                  1446 non-null   float64
 11  rean

## Initial Observations

The integrated dataset contains 1,456 weekly observations collected from two geographic locations and includes both disease incidence and environmental variables.

The dataset incorporates temperature, precipitation, humidity, and vegetation indicators that are known to influence mosquito breeding and dengue transmission dynamics.

No missing values were detected in the merged identifiers or disease incidence variables during the integration process.

The next stage of the project will explore temporal disease patterns, seasonal fluctuations, and geographic differences in dengue activity.